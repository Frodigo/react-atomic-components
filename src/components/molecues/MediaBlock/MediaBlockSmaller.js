import React from 'react';
import PropTypes from 'prop-types';

import Heading from '../../atoms/Text/Heading';

import classes from './MediaBlockSmaller.module.css';

const MediaBlockSmaller = (props) => {
    const {
        imageWidth,
        imageHeight,
        imageUrl,
        subtitle,
        title,
        text,
        roundedImage,
        ...rest
    } = props;
    const mayDisplaySubtitle = subtitle ? (
        <span className={classes.subtitle}>{subtitle}</span>
    ) : null;

    const mediaBlockImage =
        imageUrl && imageUrl ? (
            <figure
                className={
                    roundedImage ? classes.roundedFigure : classes.figure
                }
            >
                <img
                    src={imageUrl}
                    alt={title}
                    width={imageWidth}
                    height={imageHeight}
                />
            </figure>
        ) : null;

    return (
        <div className={classes.mediaBlockSmaller} {...rest}>
            {mediaBlockImage}
            <section className={classes.content}>
                {mayDisplaySubtitle}
                <Heading type="h3">{title}</Heading>
                {text}
            </section>
        </div>
    );
};

MediaBlockSmaller.propTypes = {
    imageHeight: PropTypes.number,
    imageUrl: PropTypes.string,
    imageWidth: PropTypes.number,
    roundedImage: PropTypes.bool,
    subtitle: PropTypes.string,
    text: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
};

export default MediaBlockSmaller;
