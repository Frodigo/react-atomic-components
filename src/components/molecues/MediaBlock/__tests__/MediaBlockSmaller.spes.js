import React from 'react';
import { render } from '@testing-library/react';

import MediaBlockSmaller from '../MediaBlockSmaller';

describe('MediaBlockSmaller should be rendered', () => {
    const props = {
        imageUrl: 'https://via.placeholder.com/290x240',
        subtitle: '05/2020 - 10/2020',
        text: 'Apple',
        title: 'Front end developer'
    };

    test('MediaBlockSmaller renders correctly', () => {
        const { container } = render(<MediaBlockSmaller {...props} />);

        expect(container).toMatchSnapshot();
    });
});
