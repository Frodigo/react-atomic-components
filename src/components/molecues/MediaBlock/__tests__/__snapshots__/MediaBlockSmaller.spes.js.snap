// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaBlockSmaller should be rendered MediaBlockSmaller renders correctly 1`] = `
<div>
  <div
    class="mediaBlockSmaller"
  >
    <figure
      class="figure"
    >
      <img
        alt="Front end developer"
        src="https://via.placeholder.com/290x240"
      />
    </figure>
    <section
      class="content"
    >
      <span
        class="subtitle"
      >
        05/2020 - 10/2020
      </span>
      <h3
        class=""
      >
        Front end developer
      </h3>
      Apple
    </section>
  </div>
</div>
`;
